web/src/main/java/org/openmrs/web/filter/util/CustomResourceLoader.java

```diff
@@ -92,7 +92,10 @@ private Locale parseLocaleFrom(String filename, String basename) {
 		Locale result = null;
 		
 		if (filename.startsWith(basename)) {
-			filename = filename.substring(basename.length());
+			
+			String tempfilename = filename;
+			tempfilename = tempfilename.substring(basename.length());
+			filename = tempfilename;
```

Comment:
The issue explicitly says not to reassign method parameters. But you are reassigning the `filename` method parameter like:

    filename = tempfilename;

So what you have to do imo is to declare the temporary variable just before the if block and then reuse it.

    tempfilename = filename.substring(basename.length());